guava/src/com/google/common/primitives/Bytes.java

```diff
@@ -152,6 +152,43 @@ private static int lastIndexOf(byte[] array, byte target, int start, int end) {
   }
 
   /**
+   * Returns the first position of the last occurrence of the specified {@code
+   * target} within {@code array}, or {@code -1} if there is no such occurrence.
+   *
+   * <p>More formally, returns the greatest index {@code i} such that
+   * {@code Arrays.copyOfRange(array, i, i + target.length)} contains exactly the same elements as
+   * {@code target}.
+   *
+   * @param array the array to search for the sequence {@code target}
+   * @param target the array to search for as a sub-sequence of {@code array}
+   */
+  public static int lastIndexOf(byte[] array, byte[] target) {
+    checkNotNull(array, "array");
+    checkNotNull(target, "target");
+    if (target.length == 0) {
+      return array.length - 1;
```

Comment:
Why has this chosen to return `array.length - 1` and not `array.length`?

The former essentially represents the 0-length subarray before the last element, not after it. The latter makes more semantical sense to me for that reason.